AudioKit/Core Classes/AKManager.m

```diff
@@ -116,15 +112,20 @@ - (instancetype)init {
         _isBatching = NO;
         
         _orchestra = [[AKOrchestra alloc] init];
+        
+        NSString *inputOption = [NSString stringWithFormat:@"-i %@", AKSettings.shared.audioInput];
+#ifdef TRAVIS_CI
+        inputOption = @"";
+#endif
 
         _options = [NSString stringWithFormat:
                     @"-o %@           ; Write sound to the host audio output\n"
                     "--expression-opt ; Enable expression optimizations\n"
                     "-m0              ; Print raw amplitudes\n"
-                    "-i %@            ; Request sound from the host audio input device",
-                    AKSettings.shared.audioOutput, AKSettings.shared.audioInput];
+                    "%@\n",
+                    AKSettings.shared.audioOutput, inputOption];
         
-        _csdFile = [NSString stringWithFormat:@"%@/AudioKit-%@.csd", NSTemporaryDirectory(), @(getpid())];
+        _csdFile = [NSString stringWithFormat:@"%@/AudioKit.csd", NSTemporaryDirectory()];
```

Comment:
Well the right way to do this, considering these are temporary files, would be to make sure that they are removed upon program termination. :)

Arguably this is not too much of a problem on iOS where this is all within the app sandbox, but could be a security issue on the Mac.

--
Stéphane Peter
megastep@megastep.org




> On May 24, 2015, at 1:43 AM, Aurelius Prochazka <notifications@github.com> wrote:
> 
> In AudioKit/Core Classes/AKManager.m <https://github.com/audiokit/AudioKit/pull/158#discussion_r30951034>:
> 
> >          
> > -        _csdFile = [NSString stringWithFormat:@"%@/AudioKit-%@.csd", NSTemporaryDirectory(), @(getpid())];
> > +        _csdFile = [NSString stringWithFormat:@"%@/AudioKit.csd", NSTemporaryDirectory()];
> AudioKit produces the same file over and over again, it's never different for different builds, so I got sick of their being a lot of files in that directory. I have no strong opinion, so if you want it back, I can put it back.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/audiokit/AudioKit/pull/158/files#r30951034>.
>